client/models/common.model.ts

```diff
@@ -0,0 +1,51 @@
+export type Table =
+  | "users"
+  | "user_category"
+  | "jobs"
+  | "job_category"
+  | "job_applications"
+  | "gigs"
+  | "gig_category"
+  | "gig_applications"
+  | "categories"
+  | "company_members"
+  | "links";
+
+export type WithTimestamp = {
+  created_at: Date;
+  updated_at: Date;
+  deleted_at?: Date;
+};
+
+export type CreatedBy = {
+  created_by: string;
+};
+
+export type PrimaryKey = {
+  id: string;
+};
+
+export type Application = {
+  pitch?: string;
+  content?: string;
+  status?: ApplicationStatus;
+};
+
+/**
+ * Status indicator for job/gig applications.
+ *
+ * Default status should always be "Draft"
+ */
+export type ApplicationStatus =
+  | "Draft"
+  | "Pending"
+  | "Viewed"
+  | "Ready for Interview"
+  | "Done";
+
+/**
+ * Status for job/gig postings.
+ *
+ * Default status should always be "Draft"
+ */
+export type PostingStatus = "Draft" | "Published" | "Closed";
```

Comment:
How would you feel about defining these types of types (lol) as `enum` instead of `type`?

```ts
export enum PostingStatus {
  Draft = "Draft",
  Published = "Published",
  Closed = "Closed",
}
```

I tend to like the DX around `enum` more for two reasons:

1. We're not just passing a bunch of raw strings around.
2. Kinda related to 1, but it's really easy to change the key and/or value of one of the keys. 

- Renaming a key in VSCode/`neovim`/etc. renames it everywhere. 
- Changing the value only requires us to change it in the `enum` definition and that change is now applied everywhere the `enum` is used.
  - If it's a type, we have to manually update that string anywhere it's used. That might not always be easy for strings used by multiple types e.g. `ApplicationStatus` and `PostingStatus` both have `"Draft"` so we couldn't just find+replace all instances of `"Draft"`

The one downside to `enum` is that you _have_ to `import` and use the `enum` to refer to one of its values:

```ts
let postingStatus: PostingStatus = PostingStatus.Draft // works
postingStatus = "Draft" // type error
```

Reply:
> Kinda related to 1, but it's really easy to change the key and/or value of one of the keys.

Totally agree on that one! Thanks for making me realize that :) 

By the way I kinda made it as a type and using string, just because of the fact that I can create types on strings in TypeScript and it was not possible to do it in other languages. So that's why that's that 😁  

Though other than that I prefer enums

Reply:
@with-heart  I also want to know for enums like these on do we prefer on PascalCasing or do we go ALL_CAPS_CASING? I want to know what are your thoughts on that on when do we use the ALL_CAPS_CASING. Or we're not actually using ALL_CAPS_CASING for enums but rather only for constants?

Reply:
TypeScript standard is pascal for `enum` names and keys: `ApplicationStatus.ReadyForInterview`